### PR TITLE
Regex for promocode embedded inside quotes after the keyword "code"

### PR DESCRIPTION
**Pull request by margaretchan**
ex. "Use my code "20OFF" for 20% off."

Accounts for both single and double quotes and allows all letters (upper and lowercase), numbers, and symbols in the matched promocode.

### CODE REVIEW

File: src/test/java/com/google/step/youtube/DescriptionParserTest.java [hunk 1]
```diff
@@ -195,7 +204,166 @@ public void codeNoQuotes_multipleMatches() {
...
+    @Test
+    public void codeWithQuotes_lazyMatching() {
+        String desc = "Get %15 off Rhino Skin, Unparallel shoes and much more at "
+            + "https://darkventures.co.uk/shop with offer code \"bobat\"15\"";
+
+        List<String> actual = DescriptionParser.findMatches(CODE_WITH_QUOTES_PATTERN, desc);
+        assertThat(actual, equalTo(Arrays.asList("bobat")));
+    }
+
+    /** CODE_WITH_QUOTES_PATTERN: also allow single quotes */
```
Comment (josefj1519) [inline src/test/java/com/google/step/youtube/DescriptionParserTest.java]: Some of these function names are self explanatory.  I'm not sure if comments are needed.

File: src/test/java/com/google/step/youtube/DescriptionParserTest.java [hunk 2]
```diff
@@ -56,14 +65,14 @@ public void parse_codeNoQuotes() {
 
     private static final Pattern CODE_NO_QUOTES_PATTERN = DescriptionParser.Patterns.CODE_NO_QUOTES.getPattern();
 
-    /** CODE_NO_QUOTES: standard case - 1 space, end of string, all caps, letters and numbers */
+    /** CODE_NO_QUOTES: letters and numbers */
```
Comment (josefj1519) [inline src/test/java/com/google/step/youtube/DescriptionParserTest.java]: Function and comment are the same.

File: src/test/java/com/google/step/youtube/DescriptionParserTest.java (hunk 1 above)
Comment (margaretchan) [reply src/test/java/com/google/step/youtube/DescriptionParserTest.java]: Removed unnecessary comments.

File: src/test/java/com/google/step/youtube/DescriptionParserTest.java (hunk 2 above)
Comment (margaretchan) [reply src/test/java/com/google/step/youtube/DescriptionParserTest.java]: Removed redundant comments.

**Review by josefj1519 — APPROVED** [review]

**Review by dallen68 — APPROVED** [review]

**Review by LiRq95 — APPROVED** [review]

**Pull request merged**